c++/src/CpuInfoUtil.hh

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ORC_CPUINFOUTIL_HH
+#define ORC_CPUINFOUTIL_HH
+
+#include <cstdint>
+#include <memory>
+#include <string>
+
+namespace orc {
+
+  /**
+   * CpuInfo is an interface to query for cpu information at runtime.  The caller can
+   * ask for the sizes of the caches and what hardware features are supported.
+   * On Linux, this information is pulled from a couple of sys files (/proc/cpuinfo and
+   * /sys/devices)
+   */
+  class CpuInfo {
```

Comment:
This file comes mostly from https://github.com/apache/arrow/blob/main/cpp/src/arrow/util/cpu_info.h and it contains many unused code path. Should we tailor it to match our need or add a comment to explicitly saying we are borrowing the code from there? I am not sure about this risk @dongjoon-hyun @williamhyun

Reply:
Already added a comment about this code borrowed from Apache/arrow.
https://github.com/wpleonardo/orc/blob/fe5b6c7a29721bb5d8c4699a0b072d64555d600d/c%2B%2B/src/CpuInfoUtil.hh#L19
Do we have any license issues about that?

Reply:
Both are Apache-2.0 license, I think it is OK.